### PR TITLE
Vhost and loation proxy_cache_key and proxy_cache_use_stale

### PR DESCRIPTION
**Pull request by jacobmw**

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -66,6 +66,7 @@
   $proxy_buffers                  = '32 4k',
   $proxy_buffer_size              = '8k',
   $proxy_cache_inactive           = '20m',
+  $proxy_cache_key                = '$scheme$proxy_host$request_uri',
```
Comment (3flex) [inline manifests/config.pp]: I'm not seeing where this is referenced from... can it be removed?

**Comment by 3flex** [comment]
Apart from my comment, looks good. Can you address that feedback, then squash the submission down to one commit? Then we can get it merged.

If you're not sure how to do this, there's a useful guide here: https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history

File: manifests/config.pp (hunk 1 above)
Comment (jacobmw) [inline manifests/config.pp]: $scheme$proxy_host$request_uri is the actual default value:
http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_key

Comment (3flex) [inline manifests/config.pp]: Yep, but the nginx::config::proxy_cache_key parameter you created here isn't referenced anywhere else.

**Comment by jacobmw** [comment]
I see what you mean about the proxy_cache_key.  I've removed it and attempted to squash my submission.  Please let me know if there's anything else I need to do.

**Pull request merged**